lib/vault/client.rb

```diff
@@ -387,19 +388,20 @@ def success(response)
     #   the response object from the request
     def error(response)
       if response.body && response.body.match("missing client token")
-        raise MissingTokenError
-      end
-
-      # Use the correct exception class
-      case response
-      when Net::HTTPPreconditionFailed
-        raise MissingRequiredStateError.new
-      when Net::HTTPClientError
+        # this behavior changed in Vault 1.10
```

Comment:
Can you elaborate on this a bit more? Someone who's not looking at this particular diff/PR might not know what this change is referring to.

Reply:
:+1: